### PR TITLE
(De)Serializing the NodeModel with IResourceStorageFacade is not working if transient features involved

### LINKED ISSUE #65: (De)Serializing the NodeModel with IResourceStorageFacade is not working if transient features involved
We try to serialize a `BatchLinkableResource` (we use xbase)  using `BatchLinkableResourceStorageFacade`. this is not working.

The problem is that our model contains `XAssignment`s. These have containment references that are transient so EMF wont serialize them.

`org.eclipse.xtext.nodemodel.serialization.SerializationUtil.fillIdToEObjectMap(EObject, List<EObject>)` does not care about this.

### PR DESCRIPTION
**Pull request by cdietrich**
Fixes #65

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
please review in conjunction with eclipse/xtext-xtend#22 and eclipse/xtext-extras#28/

File: org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/serialization/SerializationUtil.java [hunk 1]
```diff
@@ -48,7 +48,9 @@ public static void fillIdToEObjectMap(Resource resource, List<EObject> map) {
 	}
 
 	public static void fillIdToEObjectMap(EObject eObject, List<EObject> map) {
-		map.add(eObject);
+		if (eObject.eContainingFeature() == null || !eObject.eContainingFeature().isTransient()) {
+			map.add(eObject);
+		}
```
Comment (svenefftinge) [inline org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/serialization/SerializationUtil.java]: don't we also want to skip the contents of a transient object?

Comment (cdietrich) [inline org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/serialization/SerializationUtil.java]: Good question. I'll have a look

Comment (cdietrich) [inline org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/serialization/SerializationUtil.java]: i changed it. emf does not serialize non  transient children of transient objects

Comment (svenefftinge) [inline org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/serialization/SerializationUtil.java]: Children of transient object become transient automatically. They are after all bound to the lifecycle of their parent.

**Pull request merged**